### ISSUE TITLE
nil globals

### ISSUE THREAD

**Issue by eridal** [jamesmarlowe/lua-resty-aws#4]
Hi,

I'm trying to use the library and found a couple of places where there was a missing global reference. 

Those were easily fixed by simply requiring the module into a local variable, but this one is a bit tricky and I'm not yet able to find out.

        > aws.lua:71: attempt to index global 'str' (a nil value)

The code is doing:

```lua
69: local sha256 = resty_sha256:new()
70: sha256:update(_rest_information.body)
71: local _payload_hash = str.to_hex(sha256:final())
```

There are, also, other references to `str` below that line, which may indicate that it's a global reference .. to a string utility?

Finally it's worth saying that I don't have much experience working with lua, so this error may be just me missing something.

Are these references supposed to work out of the box? Am I missing some config to make the module work?

**Comment by eridal** [comment]
Ok, so I've replaced the `to_hex` with a manualy coded function, and now it's failing becuase of the missiing `crypto_hmac`

> https://github.com/jamesmarlowe/lua-resty-aws/blob/3cd419392c6a1e8fabd6d5c899510fa482ad21a6/lib/resty/aws.lua#L83

Are these supposed to be globals? Should I just install any some sort of dependency to make this work?

**Comment by jamesmarlowe** [comment]
You are correct, when I developed the library 2 years ago it was in conjunction with another repository which was setting up globals.

The two you pointed out are (There may be more)
```
str = require "resty.string"
crypto_hmac = require "crypto.hmac"
```
I believe they're from https://github.com/openresty/lua-resty-string and  https://github.com/starius/luacrypto though I'm sure there have been relevant updates or changes in that time.

**Comment by eridal** [comment]
Hey there,

Thanks for you imput! I've just created the PR https://github.com/jamesmarlowe/lua-resty-aws/pull/5 to address the fix to get `kinesis_streams` working
